ql/src/java/org/apache/hadoop/hive/ql/ddl/table/storage/compact/AlterTableCompactOperation.java

```diff
@@ -99,7 +99,7 @@ private void waitForCompactionToFinish(CompactionResponse resp) throws HiveExcep
     wait: while (true) {
       //double wait time until 5min
       waitTimeMs = waitTimeMs*2;
-      waitTimeMs = Math.max(waitTimeMs, waitTimeOut);
+      waitTimeMs = Math.min(waitTimeMs, waitTimeOut);
```

Comment:
Why do we need this change? Per the documentation, the minimum wait timeout is 2000 ms

Reply:
default is 300000ms - 5 min

Reply:
I've got your point.